.evergreen/serverless/delete-instance.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
```

Comment:
I used `shellcheck -s sh` to verify that neither of these scripts used bash syntax. Note that `get-instance.sh` is still invoked with `bash` from `create-instance.sh`. I'm open to removing that, but it seems harmless to leave as-is.

Reply:
Personally I would prefer encouraging scripts to support and use Bash only, rather than the limited features of an inconsistent `sh` shell (Bourne? Bash? Dash?).

Reply:
I'd prefer leaving these as bash. It may not need bash syntax now but it might in the future and leaving it as bash avoids churn. Bash is also more consistent since it seems like we are moving towards requiring bash in all scripts.